tests/Unit/Options/Test_Auto.cpp

```diff
@@ -36,6 +36,17 @@ void test_class() noexcept {
     CHECK((extracted and *extracted and **extracted == 3));
   }
 
+  {
+    struct A {};
+    struct B {
+      // NOLINTNEXTLINE(google-explicit-constructor)
+      B(A /*unused*/) noexcept {}
+    };
+
+    // Test that converting the contained class compiles.
+    (void)static_cast<std::optional<B>>(Options::Auto<A>{});
```

Comment:
Will this also work:

```cpp
static_cast<std::optional<std::unique_ptr<Base>>>(Options::Auto<std::unique_ptr<Derived>>{});
```

Just checking, since this is the use case I had in mind. Specifically, factory-creating an option using the `creatable_classes` mechanism, but taking it in the constructor as an abstract base class that doesn't know the `creatable_classes`.

Reply:
I added an explicit test, since it is the case we care about most.